src/main/java/seedu/address/model/person/Log.java

```diff
@@ -0,0 +1,91 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Objects;
+
+/**
+ * Represents a note or log tied to a Person in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTitle(String)}
+ */
+public class Log {
+
+    // default values
+    public static final String DEFAULT_NO_DESCRIPTION = "";
+
+    // constraints
+    public static final int TITLE_LENGTH_CONSTRAINT = 50;
+    public static final String TITLE_CONSTRAINTS = "Titles of logs must satisfy:\n"
+            + "1. Not be trivial (i.e. not empty or contain only spaces\n"
+            + "2. Be at most " + Log.TITLE_LENGTH_CONSTRAINT + " characters long. "
+            + "This is because of display limitations.";
+
+    // immutable attributes
+    private final String title;
+    private final String description;
+
+
+    /**
+     * Constructs a Log object.
+     *
+     * @param title       Title of the log.
+     * @param description Description tied to the log. Can be null.
+     */
+    public Log(String title, String description) {
+        requireNonNull(title);
+        checkArgument(isValidTitle(title), Log.TITLE_CONSTRAINTS);
+        this.title = title;
+        this.description = Objects.requireNonNullElse(description, Log.DEFAULT_NO_DESCRIPTION);
+    }
+
+    /**
+     * Returns true if a given string is a valid title.
+     */
+    public static boolean isValidTitle(String title) {
```

Comment:
Good effort in adding a check here to ensure the validity of the title of a log!